fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java

```diff
@@ -70,32 +85,89 @@ public void onStop() {
     }
   }
 
-  @Override
-  public boolean onOptionsItemSelected(@NonNull MenuItem menuItem) {
-    if (menuItem.getItemId() == android.R.id.home) {
-      Fragment fragment =
-          getSupportFragmentManager().findFragmentById(R.id.fragment_container_lao_detail);
-      if (fragment instanceof LaoDetailFragment) {
-        startActivity(HomeActivity.newIntent(this));
-      } else {
-        if (viewModel.getCurrentTab().getValue() == LaoTab.EVENTS) {
-          // On reselection the navigation is supposed to do nothing to prevent loops, so we
-          // manually change the fragment
-          openEventsTab();
-        } else {
-          viewModel.setCurrentTab(LaoTab.EVENTS);
-        }
-      }
-      return true;
-    }
-    return super.onOptionsItemSelected(menuItem);
+  private void setupTopAppBar() {
+    viewModel.getPageTitle().observe(this, binding.laoTopAppBar::setTitle);
+
+    // We subscribe to lao updates to display name iff the event list is displayed
+    // This is far from elegant but I don't see a better way
+    viewModel
+        .getCurrentLao()
+        .observe(
+            this,
+            laoView -> {
+              Fragment fragment =
+                  getSupportFragmentManager().findFragmentById(R.id.fragment_container_lao_detail);
+              if (fragment instanceof LaoDetailFragment) {
+                viewModel.setPageTitle(laoView.getName());
+              }
+            });
+
+    binding.laoTopAppBar.setNavigationOnClickListener(
+        v -> {
+          Fragment fragment =
+              getSupportFragmentManager().findFragmentById(R.id.fragment_container_lao_detail);
+          if (fragment instanceof LaoDetailFragment) {
+            startActivity(HomeActivity.newIntent(this));
+          } else {
+            if (viewModel.getCurrentTab().getValue() == LaoTab.EVENTS) {
+              // On reselection the navigation is supposed to do nothing to prevent loops, so we
+              // manually change the fragment
+              openEventsTab();
+            } else {
+              viewModel.setCurrentTab(LaoTab.EVENTS);
+            }
+          }
```

Comment:
This is for the back arrow top yes. The problem this solves is the following. The reselection is disabled to prevent loops and if you are in a fragment that is not on tabs (e.g. rc creation) you might still want to go back to the event list with the back arrow. If you went into said fragment from the event tab (which is almost always the case) the reselection disabling will prevent you from going back

This is not very elegant but it was already in the code base so I don't this PR would be the place to fix it.
Though if you have a better idea about how to handle this we can :)